packages/services/proxy-worker/src/handler.ts

```diff
@@ -0,0 +1,46 @@
+import { isSignatureValid } from './auth';
+import { parseIncomingRequest } from './models';
+
+/**
+ * gatherResponse awaits and returns a response body as a string.
+ */
+async function gatherResponse(response: Response) {
+  const contentType = response.headers.get('content-type') || '';
+
+  if (contentType.includes('application/json')) {
+    return JSON.stringify(await response.json());
+  } else if (contentType.includes('application/text')) {
+    return response.text();
+  } else if (contentType.includes('text/html')) {
+    return response.text();
+  } else {
+    return response.text();
+  }
+}
+
+export async function handleRequest(request: Request, keyValidator: typeof isSignatureValid) {
+  const parsedRequest = await parseIncomingRequest(request, keyValidator);
+
+  if ('error' in parsedRequest) {
+    return parsedRequest.error;
+  }
+
+  const init =
+    parsedRequest.method === 'GET'
+      ? {
+          method: 'GET',
+          headers: parsedRequest.headers,
+        }
+      : {
+          method: 'POST',
+          body: parsedRequest.body,
+          headers: parsedRequest.headers,
+        };
+  const response = await fetch(parsedRequest.url, init);
+  const text = await gatherResponse(response);
+  return new Response(text, {
```

Comment:
```suggestion
  return new Response(response.body, {
```
We can pass the body as-is.